### ISSUE TITLE
More robust support for plugins/sections/blocks

### ISSUE THREAD

**Issue by strafe** [cmacrae/spacebar#58]
The new shell config options are amazing, being able to add custom output means I can finally stop using Uebersicht so thank you!

I'd love to see the ability to stack multiple commands from the left and right sides rather than just a single command for each, this has tons of benefits. Each plugin/block would provide an icon, command and update interval (on click actions would be incredible too if that's even possible).

**Comment by cmacrae** [comment]
Hey @strafe 👋🏻 Glad you're liking the new features!

Yes, I'd been planning to implement what you've described here :)

I'll see what I can put together soon

**Comment by itsdawei** [comment]
We should also open an issue for people to share their custom shell scripts! It is difficult for some people with less shell scripting experience (me included) to create their own custom plugin.

**Comment by cmacrae** [comment]
Yeah! I've been thinking I'll enable [GitHub Discussions](https://docs.github.com/en/discussions) so we have a forum :)  
I'll set it up once I'm done wrapping up the feature I'm working on

**Comment by cmacrae** [comment]
@itsdawei I created a discussion here :)
https://github.com/cmacrae/spacebar/discussions/70

**Comment by cmacrae** [comment]
A sneak peek of this in the rewrite:  
![image](https://user-images.githubusercontent.com/3392199/116219498-96d32700-a743-11eb-831f-6284782dffed.png)

**Comment by strafe** [comment]
> A sneak peek of this in the rewrite:
> ![image](https://user-images.githubusercontent.com/3392199/116219498-96d32700-a743-11eb-831f-6284782dffed.png)

Great job on this!

**Comment by cmacrae** [comment]
Thanks @strafe !  
Yeah, sure 👍 Separators should be easy enough :)  
Forgive the silly question, but where are the separators in the image you've shared? Is it just the blank space that you're referring to?

**Comment by strafe** [comment]
No problem, circled them and zoomed in.
![CleanShot 2021-04-27 at 16 52 44@2x](https://user-images.githubusercontent.com/15663890/116273153-6b6b2f00-a779-11eb-9594-df472e053012.png)

**Comment by cmacrae** [comment]
Ah! I should've clicked to open the image 😅 No problem, I can definitely implement that 👍
